alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunner.kt

```diff
@@ -292,7 +298,14 @@ class MonitorRunner(
                         val searchResponse: SearchResponse = client.suspendUntil { client.search(searchRequest, it) }
                         results += searchResponse.convertToMap()
                     }
-                    else -> {
+                    is HttpInput -> {
+                        val httpClient = HttpInputClient()
```

Comment:
Avoid creating a new client on each input.  Use a single instance.

Reply:
Decided to not to have a single instance of `HttpInputClient` because it needs to be recreated every time when closed, and it would not be good to leave a single instance running the whole time. This is due to the way `HttpAsyncClient` works.

Reply:
What do you mean by "running the whole time"? The client should only be closed when the application shuts down - it doesn't consume any resources (other than a few bytes for the memory when there are no requests running).  The example in the QuickStart guide also uses a single client for making multiple requests: https://hc.apache.org/httpcomponents-asyncclient-dev/quickstart.html. 

Creating a client is an expensive operation and should only be done once.

Reply:
I can create one client to reuse, and call `execute(request)` for each request. But then I will not be closing the client anywhere in `MonitorRunner`. Do you have any suggestions on where to put the`client.close()`?

Reply:
Either stop it in `MonitorRunner.doStop` or you can make `HttpInputClient` a `LifecycleComponent`, register it in `MonitoringPlugin.createComponents` and have the ES server start and stop it.

Reply:
Added `close()` in `MonitorRunner.doStop()` function. Modified the creation and `start()` of client to `doStart()` and `init` respectively.